lib/service/supervisor.go

```diff
@@ -511,6 +499,30 @@ func (s *LocalSupervisor) ListenForEvents(ctx context.Context, name string, even
 	s.eventWaiters[name] = append(s.eventWaiters[name], waiter)
 }
 
+func (s *LocalSupervisor) ListenForNewEvent(name string) (<-chan Event, context.CancelFunc) {
+	s.Lock()
+	defer s.Unlock()
+
+	ctx, cancel := context.WithCancel(s.closeContext)
+
+	waiterC := make(chan Event)
+	waiter := &waiter{eventC: waiterC, context: ctx}
+	s.eventWaiters[name] = append(s.eventWaiters[name], waiter)
+
+	outputC := make(chan Event, 1)
+	go func() {
+		defer close(outputC)
+		defer cancel()
+		select {
+		case <-ctx.Done():
+		case e := <-waiterC:
+			outputC <- e
+		}
+	}()
+
+	return outputC, cancel
+}
```

Comment:
for a more consistent signature and 1 fewer goroutine
```suggestion
func (s *LocalSupervisor) ListenForNewEvent(ctx context.Context, name string, eventC chan<- Event) {
	s.Lock()
	defer s.Unlock()

	waiter := &waiter{eventC: eventC, context: ctx}
	s.eventWaiters[name] = append(s.eventWaiters[name], waiter)
	return
}
```

Reply:
I was thinking of having this be more of a utility method but seeing as we use it in literally one helper function in tests, I agree with just making it `ListenForNewEvents` as a variant of `ListenForEvents` that doesn't immediately send the first value in.